languages/tree-sitter-stack-graphs-java/src/stack-graphs.tsg

```diff
@@ -355,46 +866,136 @@ attribute node_symbol = node            => symbol = (source-text node), source_n
   edge expr_ref__typeof -> expr_ref
 }
 
-(local_variable_declaration
-  type: (type_identifier) @type
-  declarator: (variable_declarator
-    name: (identifier) @identifier
-    value: (object_creation_expression
-      type: (type_identifier) @object)
-  )) @local_variable {
-  node @object.value
-  attr (@object.value) node_reference = @object
-  node @type.value
-  attr(@type.value) node_reference = @type
-  edge @type.value -> @local_variable.after_scope
-  edge @local_variable.after_scope -> @object.value
-  edge @object.value -> @local_variable.after_scope
+(method_reference . (_) @lhs) @this {
+  edge @lhs.lexical_scope -> @this.lexical_scope
+}
+
+(method_reference . (_) @lhs (identifier) @id) @this {
+  node @this.member
+
+  node expr_ref
+  attr (expr_ref) node_reference = @id
+  edge expr_ref -> @this.member
+
+  attr (@this.member) push_symbol = "."
+  edge @this.member -> @lhs.type
+
+  node expr_ref__typeof
+  ; edge @this.type -> expr_ref__typeof ; FIXME: function types
```

Comment:
This one is pretty interesting. I don't actually know what API is available for function types; is there a `Fn<(A, B), C>` class squirrelled away somewhere like C# does? Inquiring minds want to know.